src/realm.js

```diff
@@ -31,7 +32,7 @@ Realm.prototype = {
   add (key, config, options) {
     let domain = createOrClone(config, options, this.repo)
 
-    this.domains.push([key, domain])
+    this.domains.push([castPath(key), domain])
```

Comment:
This is a perf opt. We avoid re-casting string keys every time we dispatch. Pretty nice bump.